app/helpers/vaccine_query_helper.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+# Helper methods for vaccine-related queries
+module VaccineQueryHelper
+  # Validates the request params for the index action
+  def validate_table_query(params)
+    permitted_params = params.permit(:entries, :page, :search, :order, :direction)
+
+    patient_id = params.require(:patient_id)&.to_i
+    search_text = permitted_params[:search]
+    sort_order = permitted_params[:order]
+    sort_direction = permitted_params[:direction]
+    # Assume default values if pagination data is not explicitly specified
+    entries = permitted_params[:entries]&.to_i || 10
+    page = permitted_params[:page]&.to_i || 0
+
+    if sort_direction.present? && !%w[asc desc].include?(sort_direction)
+      error_message = "Unable to sort column in specified direction in request: '#{sort_direction}'"
+      raise StandardError, error_message
```

Comment:
Yeah I actually created a custom error class at first, just like we already do [here](https://github.com/SaraAlert/SaraAlert/blob/master/app/helpers/patient_query_helper.rb#L587), but given all I wanted to make "custom" about it was insert a special message, one of the linters didn't like it since it was considered a "useless" class since it functioned exactly the same as `StandardError`. I'm happy to go that route again if we think it's worth it.